components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2715,4 +2670,117 @@ private int addAuthenticator(Connection conn, int tenantId, String idpName,
         return authenticatorId;
     }
 
+    /**
+     * Read application role permissions for a given application name
+     *
+     * @param connection      Database connection
+     * @param applicationName Application name
+     * @return Map of key value pairs. key is UM table id and value is permission
+     * @throws SQLException
+     */
+    private Map<String, String> readApplicationPermissions(Connection connection, String applicationName) throws SQLException {
+        PreparedStatement readPermissions = null;
+        ResultSet resultSet = null;
+        Map<String, String> permissions = new HashMap<>();
+        try {
+            readPermissions = connection.prepareStatement(ApplicationMgtDBQueries.LOAD_UM_PERMISSIONS);
+            readPermissions.setString(1, "%" + ApplicationMgtUtil.getApplicationPermissionPath() + "%");
+            resultSet = readPermissions.executeQuery();
+            while (resultSet.next()) {
+                String UM_ID = resultSet.getString(1);
+                String permission = resultSet.getString(2);
+                if (permission.contains(ApplicationMgtUtil.getApplicationPermissionPath() +
+                        ApplicationMgtUtil.PATH_CONSTANT + applicationName.toLowerCase())) {
+                    permissions.put(UM_ID, permission);
+                }
+            }
+        } finally {
+            IdentityDatabaseUtil.closeResultSet(resultSet);
+            IdentityDatabaseUtil.closeStatement(readPermissions);
+        }
+        return permissions;
+    }
+
+    /**
+     * Update the permission path for a given id
+     *
+     * @param connection    Database connection
+     * @param um_id         Id
+     * @param newPermission New permission path value
+     * @throws SQLException
+     */
+    private void updatePermissionPath(Connection connection, String um_id, String newPermission) throws SQLException {
+        PreparedStatement updatePermission = null;
+        try {
+            updatePermission = connection.prepareStatement(ApplicationMgtDBQueries.UPDATE_SP_PERMISSIONS);
+            updatePermission.setString(1, newPermission);
+            updatePermission.setString(2, um_id);
+            updatePermission.executeUpdate();
+        } finally {
+            IdentityDatabaseUtil.closeStatement(updatePermission);
+        }
+    }
+
+    /**
+     * Get permission id for a given permission path
+     *
+     * @param connection Database connection
+     * @param permission Permission path
+     * @return Permission id
+     * @throws SQLException
+     */
+    private int getPermissionId(Connection connection, String permission) throws SQLException {
+        PreparedStatement selectQuery = null;
+        ResultSet resultSet = null;
+        int id = -1;
+        try {
+            selectQuery = connection.prepareStatement(ApplicationMgtDBQueries.LOAD_UM_PERMISSIONS_W);
```

Comment:
Variable name "selectQuery" doesn't mean anything for a prepared statement IMO. So, please use a meaningful variable name.